docs/diagrams/ArchitectureDemoSequenceDiagram.puml

```diff
@@ -0,0 +1,67 @@
+@startuml
+!include Style.puml
+
+actor User
+
+participant "duke:Duke" as Duke
+participant "ui:Ui" as Ui
+participant ":Parser" as Parser
+participant "command:AddTaskCommand" as Command
+participant "newTask:Task" as Task
+participant "taskList:TaskList" as TaskList
+participant "storage:Storage" as Storage
+
+activate Duke
+Duke -> Ui: readUserResponse()
+
+activate Ui
+User -> Ui: "add task homework -d mon"
+Ui --> Duke
+
+Duke -> Parser: parse(userResponse)
+activate Parser
+Parser --> Duke: AddTaskCommand
+deactivate Parser
+
+create Command
+Duke -> Command: AddTaskCommand(homework, Monday)
+activate Command
+
+Command --> Duke
+deactivate Command
+
+Duke -> Command: execute(ui, storage, taskList, lessonList, moduleList)
+activate Command
+
+create Task
+Command -> Task: Task(title, Monday, "-")
+
+activate Task
+Task --> Command: newTask
+deactivate Task
+
+Command -> TaskList: addTask(newTask)
+activate TaskList
+TaskList --> Command
+deactivate TaskList
+
+Command -> Storage: saveData(taskList)
+activate Storage
+Storage -> Storage: Save to task.txt
+activate Storage
+Storage --> Storage
+deactivate Storage
+Storage --> Command
+deactivate Storage
+
+Command --> Ui
+deactivate Command
+
+Ui --> User
+deactivate Ui

```

Comment:
After UI returns here, command and newTask should be scheduled to be destroyed. I suggest that "crosses" be added to show the end of their lifelines.

Reply:
Fixed.